beetsplug/lastgenre/__init__.py

```diff
@@ -179,12 +201,18 @@ def _resolve_genres(self, tags):
                     parents = [find_parents(tag, self.c14n_branches)[-1]]
 
                 tags_all += parents
-                if len(tags_all) >= count:
+                # Don't break here if we need to sort by specificity, which happens with
+                # the full tag list below
+                if not self.config['specificity'] and len(tags_all) >= count:
                     break
             tags = tags_all
 
         tags = deduplicate(tags)
 
+        # sort the tags by specificity
```

Comment:
I know the comment right below violates this :angry:, but we usually like to use normal English style for comments: i.e., start with a capital letter and end with a period.